App/Supplier/frmSupplierModal.cs

```diff
@@ -32,5 +57,248 @@ private void btnCancel_Click(object sender, EventArgs e)
+            cmbCategory.ValueMember = "id";
+            cmbCategory.DisplayMember = "name";
+
+            cmbRegion.DataSource = supplierRepository.LoadDataList("SELECT DISTINCT region.id , region.`name` FROM region;");
+            cmbRegion.ValueMember = "id";
+            cmbRegion.DisplayMember = "name";
+        }
+
+        private void cmbRegion_SelectionChangeCommitted(object sender, EventArgs e)
+        {
+            cmbProvince.DataSource = supplierRepository.LoadDataList("SELECT province.id, province.region, province.`name` FROM province INNER JOIN region ON province.region = region.id WHERE region.id =" + cmbRegion.SelectedValue.ToString() + " ORDER BY province.id");
+            cmbProvince.ValueMember = "id";
+            cmbProvince.DisplayMember = "name";
+        }
+
+        private void cmbProvince_SelectionChangeCommitted(object sender, EventArgs e)
+        {
+            cmbCity.DataSource = supplierRepository.LoadDataList("SELECT municipality.id, municipality.province, municipality.`name` FROM municipality INNER JOIN province ON municipality.province = province.id WHERE province.id =" + cmbProvince.SelectedValue.ToString() + " ORDER BY municipality.id");
+            cmbCity.ValueMember = "id";
+            cmbCity.DisplayMember = "name";
+        }
+
+        private void cmbCity_SelectionChangeCommitted(object sender, EventArgs e)
+        {
+            cmbDistrict.DataSource = supplierRepository.LoadDataList("SELECT baranggay.id, baranggay.municipality, baranggay.`name` FROM baranggay INNER JOIN municipality ON baranggay.municipality = municipality.id WHERE municipality.id =" + cmbCity.SelectedValue.ToString() + " ORDER BY baranggay.id");
+            cmbDistrict.ValueMember = "id";
+            cmbDistrict.DisplayMember = "name";
+        }
+
+        private void InitializeSelectedSupplierData()
+        {
+            InitializeComponentsData();
+            supplierRepository = new SupplierRepository();
+
+            Core.System.Data.Model.Supplier supplier = new Core.System.Data.Model.Supplier();
+            supplier = supplierRepository.FetchSupplierData(this.Id);
+            this.txtSupplierName.Text = supplier.Name;
+            cmbCategory.SelectedValue = supplier.Category.Id;
+            this.txtContactPerson.Text = supplier.Contactperson;
+            this.txtSocialNetworkID.Text = supplier.Socialnetid;
+            this.txtMobileNumber.Text = supplier.Mobilenum;
+            this.txtPhoneNumber.Text = supplier.Phonenum;
+            this.txtPhoneNumberExtension.Text = supplier.Extension;
+            this.txtEmailAddress.Text = supplier.Email;
+            cmbRegion.SelectedValue = supplier.Region.Id;
+            cmbProvince.SelectedValue = supplier.Province.Id;
+            cmbCity.SelectedValue = supplier.Municipality.Id;
+            cmbDistrict.SelectedValue = supplier.Baranggay.Id;
```

Comment:
This will not work as expected. Better populate the component first before assigning the value. Check the constructor used for updating the supplier record. It doesn't populate all dropdown components